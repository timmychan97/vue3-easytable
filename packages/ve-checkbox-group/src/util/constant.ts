export const PREFIX_CLS = 've-checkbox-group-'

export const EMIT_EVENTS = {
  ON_CHECKED_CHANGE: 'on-checked-change',
}

export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
