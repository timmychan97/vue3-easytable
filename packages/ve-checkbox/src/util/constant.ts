export const PREFIX_CLS = 've-checkbox-'

export const EMIT_EVENTS = {
  ON_CHECKED_CHANGE: 'on-checked-change',
}

export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
