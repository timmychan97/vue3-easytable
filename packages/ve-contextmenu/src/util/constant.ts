export const PREFIX_CLS = 've-contextmenu-'

export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

export const CONTEXTMENU_NODE_TYPES = {
  SEPARATOR: 'SEPARATOR',
}

export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
