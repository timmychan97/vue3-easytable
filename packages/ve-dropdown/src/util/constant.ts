export const PREFIX_CLS = 've-dropdown-'

export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
