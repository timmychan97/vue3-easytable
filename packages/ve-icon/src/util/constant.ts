export const PREFIX_CLS = 've-icon-'

export const EMIT_EVENTS = {}

export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
