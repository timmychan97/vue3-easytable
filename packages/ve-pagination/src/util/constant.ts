export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
