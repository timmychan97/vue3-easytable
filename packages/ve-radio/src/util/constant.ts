export const PREFIX_CLS = 've-radio-'

export const EMIT_EVENTS = {
  ON_RADIO_CHANGE: 'on-radio-change',
}

export const COMPS_NAME = {
  VE_RADIO: 'VeRadio',
}
