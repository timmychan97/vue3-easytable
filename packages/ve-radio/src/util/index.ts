import { PREFIX_CLS } from './constant'

export function clsName(cls: string) {
  return PREFIX_CLS + cls
}
