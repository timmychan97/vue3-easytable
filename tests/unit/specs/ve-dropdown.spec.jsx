import { expect, vi } from 'vitest'
import { later, wrapMount as mount } from '../util'
import veDropdown from '@/ve-dropdown'

describe('veDropdown', () => {
  // dropdown items
  const DROPDOWN_ITEMS = [
    { value: 0, label: 'Apple' },
    { value: 1, label: 'Orange', selected: true },
    { value: 2, label: 'Banana' },
  ]

  it('render normal', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        showOperation: true,
        confirmFilterText: 'filter',
        resetFilterText: 'reset',
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })
  it('render with radio', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        showRadio: true,
        showOperation: true,
        confirmFilterText: 'filter',
        resetFilterText: 'reset',
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('render with checkbox', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        isMultiple: true,
        showOperation: true,
        confirmFilterText: 'filter',
        resetFilterText: 'reset',
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('value props', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.findAll('.ve-dropdown-items-li').length).toBe(3)
    wrapper.unmount()
  })

  it('showOperation prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        showOperation: true,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-operation').exists()).toBe(true)
    wrapper.unmount()
  })

  it('isMultiple prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        isMultiple: true,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-items-multiple').exists()).toBe(true)
    wrapper.unmount()
  })

  it('placeholder prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        isMultiple: true,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-items-multiple').exists()).toBe(true)
    wrapper.unmount()
  })

  it('textAlign prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        textAlign: 'center',
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-items-li-a-center').exists()).toBe(
      true,
    )
    wrapper.unmount()
  })

  it('operation text prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        showOperation: true,
        confirmFilterText: 'ok',
        resetFilterText: 'cancel',
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-operation').text()).toContain('ok')
    expect(wrapper.find('.ve-dropdown-operation').text()).toContain(
      'cancel',
    )
    wrapper.unmount()
  })

  it('showRadio prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        showRadio: true,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.findAll('.ve-radio').length).toBe(3)
    wrapper.unmount()
  })

  it('isControlled prop', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        isControlled: true,
        visible: true,
      },
      slots: {
        default: `<div>click me</div>`,
      },
    })

    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(true)
    wrapper.unmount()
  })

  it('hideByItemClick prop', async () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        hideByItemClick: false,
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    wrapper.find('.btn').trigger('click')
    await later(100)
    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(true)

    wrapper.find('.ve-dropdown-items-li').trigger('click')
    await later(150)
    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(true)

    wrapper.setProps({ hideByItemClick: true })
    await later(100)
    wrapper.find('.ve-dropdown-items-li').trigger('click')
    await later(150)
    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(false)
    wrapper.unmount()
  })

  it('custom-content slot', () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
        isCustomContent: true,
      },
      slots: {
        'custom-content': `<div class="custom-content-slot">this is custom content</div>`,
      },
    })

    expect(wrapper.find('.custom-content-slot').exists()).toBe(true)
    wrapper.unmount()
  })

  it('on-item-select-change emit', async () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    wrapper.find('.btn').trigger('click')
    await later(100)

    // 改变选项
    wrapper.findAll('.ve-dropdown-items-li').at(1).trigger('click')

    await later()

    const items = [
      { value: 0, label: 'Apple', selected: false },
      { value: 1, label: 'Orange', selected: true },
      { value: 2, label: 'Banana', selected: false },
    ]

    expect(wrapper.emitted('on-item-select-change')[0]).toEqual([items])
    wrapper.unmount()
  })

  it('visible change emit', async () => {
    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: DROPDOWN_ITEMS,
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    wrapper.find('.btn').trigger('click')
    await later(100)

    wrapper.find('.btn').trigger('click')
    await later(100)

    expect(wrapper.emitted('on-dropdown-visible-change').length).toEqual(2)
    expect(wrapper.emitted('on-dropdown-visible-change')[0]).toEqual([
      true,
    ])
    expect(wrapper.emitted('on-dropdown-visible-change')[1]).toEqual([
      false,
    ])
    wrapper.unmount()
  })

  it('single select filter confirm emit', async () => {
    const items = [
      { value: 0, label: 'Apple', selected: false },
      { value: 1, label: 'Orange', selected: true },
      { value: 2, label: 'Banana', selected: false },
    ]

    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: items,
        showOperation: true,
        confirmFilterText: 'ok',
        resetFilterText: 'cancel',
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    await later()
    wrapper.find('.btn').trigger('click')
    await later()
    // 改变选项
    // wrapper.findAll('.ve-dropdown-items-li')[0].trigger('click')
    document.querySelectorAll('.ve-dropdown-items-li')[0].click()

    await later()

    // confirm btn click
    // wrapper.findAll('.ve-dropdown-operation-item')[1].trigger('click')
    document.querySelectorAll('.ve-dropdown-operation-item')[1].click()

    await later()

    expect(wrapper.emitted('on-filter-confirm').length).toEqual(1)
    expect(
      wrapper.emitted('on-filter-confirm')[0][0].find(x => x.selected)
        .label,
    ).toEqual('Apple')
    wrapper.unmount()
  })

  it('single select filter reset emit', async () => {
    const items = [
      { value: 0, label: 'Apple', selected: false },
      { value: 1, label: 'Orange', selected: true },
      { value: 2, label: 'Banana', selected: false },
    ]

    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: items,
        showOperation: true,
        confirmFilterText: 'ok',
        resetFilterText: 'cancel',
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    await later()

    wrapper.find('.btn').trigger('click')
    await later()

    // reset btn click
    // wrapper.findAll('.ve-dropdown-operation-item')[0].trigger('click')
    document.querySelectorAll('.ve-dropdown-operation-item')[0].click()

    await later()

    expect(wrapper.emitted('on-filter-reset').length).toEqual(1)
    expect(
      wrapper.emitted('on-filter-reset')[0][0].filter(x => x.selected)
        .length,
    ).toEqual(0)
    wrapper.unmount()
  })

  it('multiple select filter confirm emit', async () => {
    const items = [
      { value: 0, label: 'Apple', selected: true },
      { value: 1, label: 'Orange', selected: true },
      { value: 2, label: 'Banana', selected: false },
    ]

    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: items,
        isMultiple: true,
        showOperation: true,
        confirmFilterText: 'ok',
        resetFilterText: 'cancel',
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    await later()

    wrapper.find('.btn').trigger('click')
    await later()

    // 改变选项
    document.querySelectorAll('.ve-checkbox')[2].click()

    await later()

    // confirm btn click
    document.querySelectorAll('.ve-dropdown-operation-item')[1].click()

    await later()

    expect(wrapper.emitted('on-filter-confirm').length).toEqual(1)
    expect(
      wrapper.emitted('on-filter-confirm')[0][0].filter(x => x.selected)
        .length,
    ).toEqual(3)
    wrapper.unmount()
  })

  it('multiple select filter reset emit', async () => {
    const items = [
      { value: 0, label: 'Apple', selected: true },
      { value: 1, label: 'Orange', selected: true },
      { value: 2, label: 'Banana', selected: false },
    ]

    const wrapper = mount(veDropdown, {
      propsData: {
        modelValue: items,
        isMultiple: true,
        showOperation: true,
        confirmFilterText: 'ok',
        resetFilterText: 'cancel',
      },
      slots: {
        default: `<div class="btn">click me</div>`,
      },
    })

    await later()

    wrapper.find('.btn').trigger('click')
    await later()
    // reset btn click
    document.querySelectorAll('.ve-dropdown-operation-item')[0].click()

    await later()

    expect(wrapper.emitted('on-filter-reset').length).toEqual(1)
    expect(
      wrapper.emitted('on-filter-reset')[0][0].filter(x => x.selected)
        .length,
    ).toEqual(0)
    wrapper.unmount()
  })

  it('beforeVisibleChange callback method', async () => {
    const mockBeforeVisibleChangeFn = vi.fn()
    const wrapper = mount({
      render() {
        return (
          <ve-dropdown
            hide-by-item-click={true}
            before-visible-change={args =>
              mockBeforeVisibleChangeFn(args)}
            modelValue={DROPDOWN_ITEMS}
          >
            <div class="btn" style="color:blue;cursor: pointer;">
              点击这里
            </div>
          </ve-dropdown>
        )
      },
    })

    wrapper.find('.btn').trigger('click')
    await later(100)
    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(true)

    expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
      nextVisible: true,
    })

    wrapper.find('.ve-dropdown-items-li').trigger('click')
    await later(150)

    expect(wrapper.find('.ve-dropdown-dd-show').exists()).toBe(false)
    expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
      nextVisible: false,
    })
    wrapper.unmount()
  })
})
