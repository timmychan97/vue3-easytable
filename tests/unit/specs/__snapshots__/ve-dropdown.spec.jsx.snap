// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veDropdown > render normal 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <!---->
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Apple</a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Orange</a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Banana</a></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a class="ve-dropdown-operation-item" href="javascript:void(0)">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veDropdown > render with checkbox 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <!---->
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox" value="Apple"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Apple</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked"><input class="ve-checkbox-input" type="checkbox" value="Orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Orange</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox" value="Banana"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Banana</span></label></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a class="ve-dropdown-operation-item" href="javascript:void(0)">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veDropdown > render with radio 1`] = `
<dl class="ve-dropdown">
  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;">
      <div>click me</div>
    </a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <!---->
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Apple</span></label></a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Orange</span></label></a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left"><label class="ve-radio"><span class="ve-radio-container"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">Banana</span></label></a></li>
          </div>
          <li class="ve-dropdown-operation"><a href="javascript:void(0)" class="ve-dropdown-operation-item">reset</a><a class="ve-dropdown-operation-item" href="javascript:void(0)">filter</a></li>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;
